frontend/src/router/index.ts

```diff
@@ -13,15 +13,16 @@ export const router = createRouter({
         }
     }),
     scrollBehavior(to) {
-        // Check if the route has the `protectedLayoutMeta`
         if (to.meta?.layout === "ProtectedLayout") {
-            return {
-                top: 0,
-                behavior: "smooth",
-                el: "#proctected-layout-container",
-            };
+            const mainElement = document.querySelector(".main");
+            if (mainElement) {
+                mainElement.scrollTo({
+                    top: 0,
+                    behavior: "smooth",
+                });
+            }
+            return false; // Prevent default Vue Router scrolling behavior
```

Comment:
This is interesting, new to me.
Not sure if we need to do all of these element.scrollTo.
I saw it provides this "position" to this `scrollBehavior()` callback:
![image](https://github.com/user-attachments/assets/a0638f53-8ea4-423f-8bca-e5c8fa9fd347)
![image](https://github.com/user-attachments/assets/c4c4780b-57ea-4319-affb-2d6c8120f1b5)

Reply:
yeah we could utilize the savedPosition in the future if we need to go back to the same position on a page, currently we don't have this requirement. 
Here's the reason for using `scrollTo` 

There's a issue of this not working if we just do
```
return {
                top: 0,
                behavior: "smooth",
                el: ".main",
            };
```

because overflow-y: auto or scroll sets the scrolling behavior to the .main container, and Vue Router’s scrollBehavior typically works with the window’s scroll position, not custom scrollable elements. To make Vue Router’s scrollBehavior work with .main, we need to manually scroll the .main element instead of relying on the default window scrolling.

Reply:
No problem. Thanks for clarify this for my understanding!